lib/evm/loop.ts

```diff
@@ -75,56 +133,55 @@ module.exports = class Loop {
       // remove any logs on error
       this._result = Object.assign({}, this._result, {
         logs: [],
-        vmError: true,
         gasRefund: null,
         selfdestruct: null
       })
     }
 
     return Object.assign({}, this._result, {
-      runState: Object.assign({}, this._runState, this._result, this._runState.eei._env),
-      exception: err ? 0 : 1,
+      runState: Object.assign({}, this._runState, this._result, this._runState.eei!._env),
+      exception: err ? 0 as IsException : 1 as IsException,
       exceptionError: err,
```

Comment:
What's happening here with eei! ?

Reply:
Because `eei` is optional, typescript would give an error that it could be undefined, the `!` tells the compiler I know it isn't undefined :smile: (because it's initialized in `initRunState`). Agree this is something that can be improved.

Reply:
Ah ok, thanks.